There's also a build error for the webgpu builds:

```
CMake Error at onnxruntime.cmake:301 (target_link_options):
  target_link_options may only set INTERFACE properties on INTERFACE targets
```

--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -297,7 +297,9 @@ else()
   )
 endif()
 
-if(WIN32)
+# In a static build the onnxruntime target is an INTERFACE library, which rejects
+# the PRIVATE keyword.
+if(WIN32 AND onnxruntime_BUILD_SHARED_LIB)
   target_link_options(onnxruntime PRIVATE ${onnxruntime_DELAYLOAD_FLAGS})
 endif()
 #See: https://cmake.org/cmake/help/latest/prop_tgt/SOVERSION.html

--- a/cmake/onnxruntime_common.cmake
+++ b/cmake/onnxruntime_common.cmake
@@ -155,7 +155,9 @@ if(WIN32)
     # shell32.dll statically imports user32.dll, which is unavailable under Win32k lockdown. Delay-load
     # shell32.dll so the load-time dependency on user32.dll is deferred until the call is actually made,
     # letting onnxruntime.dll load in lockdown processes.
-    list(APPEND onnxruntime_DELAYLOAD_FLAGS "/DELAYLOAD:shell32.dll")
+    if(onnxruntime_ENABLE_DELAY_LOADING_WIN_DLLS)
+      list(APPEND onnxruntime_DELAYLOAD_FLAGS "/DELAYLOAD:shell32.dll")
+    endif()
   endif()
 endif()
 

